### ISSUE TITLE
Add optional layout syntax for match expression

### ISSUE THREAD

**Issue by momohatt** [egison/egison#156]
```
match [1, 2, 3] as list integer with { #1 :: _ -> True; #2 :: _ -> False }
```

This can be a little tricky as the starting symbol collides with the sequential pattern.

**Comment by egisatoshi** [comment]
If we support this optoinal layout syntax, I think we can remove the option to omit `|` when a pattern-match expression has only one match clause.
It will make the implementation of the parser simple.

**Comment by momohatt** [comment]
I agree that it'll make the parser simpler, but I'm not sure if it's worthwhile to add this optional layout syntax while sacrificing that feature... It depends on how often we need to write multiple match clauses in single line.
